timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureTimelockResource.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.FluentFuture;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.http.RedirectRetryTargeter;
+import com.palantir.atlasdb.timelock.api.ConjureStartTransactionsRequest;
+import com.palantir.atlasdb.timelock.api.ConjureStartTransactionsResponse;
+import com.palantir.atlasdb.timelock.api.ConjureTimelockService;
+import com.palantir.atlasdb.timelock.api.ConjureTimelockServiceEndpoints;
+import com.palantir.atlasdb.timelock.api.UndertowConjureTimelockService;
+import com.palantir.conjure.java.api.errors.QosException;
+import com.palantir.conjure.java.undertow.lib.UndertowService;
+import com.palantir.leader.NotCurrentLeaderException;
+import com.palantir.lock.impl.TooManyRequestsException;
+import com.palantir.lock.remoting.BlockingTimeoutException;
+import com.palantir.lock.v2.ImmutableStartTransactionRequestV5;
+import com.palantir.lock.v2.LeaderTime;
+import com.palantir.lock.v2.StartTransactionRequestV5;
+import com.palantir.lock.v2.StartTransactionResponseV5;
+import com.palantir.tokens.auth.AuthHeader;
+
+public final class ConjureTimelockResource implements UndertowConjureTimelockService {
```

Comment:
similar vein as comment below: comment re this class represents a new set of endpoints that are defined via conjure.

Reply:
Won't mark them as new endpoints, because they're not 'new' once this PR merges - they're the status quo. I will deprecate TimeLockResource, however - to ensure that new methods end up in here.